### PR TITLE
Fix(eos_cli_config_gen): NTP Servers are ordered like in running-config

### LINKED ISSUE #1091: Issue: ntp_servers preferred
## Issue Type

- [ ] Bug Report
- [X] EOS Configuration rendering error
- [ ] Documentation Issue Report

## Summary

`ntp_servers` preferred is set by associating the first ntp_server in the list of ntp_servers as the preferred.

This is not how eos would handle or render this config because ntp servers are placed in the config in sorted by IP.  Thus the logic of avd.eos_cli_config_gen contradicts EOS in that if you sort the ntp servers in order of IP, the first IP would automatically be set as the preferred server.

## Component Name

eos_cli_config_gen

## Steps to reproduce

```yaml
ntp_server:
  nodes:
    100.100.100.100: 
      preferred: true
    10.10.10.10: {}
```

### Expected results

I would expect this sort of data model to exist and render:
```cfg
ntp server 10.10.10.10
ntp server 100.100.100.100 prefer
```

### Actual results

```yaml
ntp_server:
  nodes:
    - 100.100.100.100:
    - 10.10.10.10
```

Renders
```cfg
ntp server 100.100.100.100 prefer
ntp server 10.10.10.10
```

### PR DESCRIPTION
**Pull request by colinmacgiolla**
## Change Summary

Generated CLI config doesn't match the switch running-config render, which has the ntp servers sorted by IP, then by name (if used)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes #1091

## Component(s) name

`arista.avd.eos_cli_gen`

## Proposed changes
No change in the datamodel (for now).

The existing datamodel assumes a list of NTP servers, with the first one being preferred. The existing behaviour renders the servers in the order they are provided, which doesn't match the outcome once configured on the device.

This change preserves the existing behaviour (1st entry in the list is preferred), but sorts the list of ntp servers

## How to test
Updated molecule testing for NTP to catch this issue

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contributing/) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

**Comment by carlbuchmann** [comment]
@colinmacgiolla - as per the comment in the issue, I suggest we update the `eos_cli_config_gen` data model.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ntp.j2 [hunk 1]
```diff
@@ -23,8 +23,9 @@
 
 | Node | Primary |
 | ---- | ------- |
-{%     for node in ntp_server.nodes | arista.avd.default([]) %}
-{%         if loop.first %}
+{%     set firstEntry = ntp_server.nodes.0 %}
+{%     for node in ntp_server.nodes | arista.avd.default([]) | arista.avd.natural_sort %}
```
Comment (titom73) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ntp.j2]: This `ntp_server.nodes | arista.avd.default([]) | arista.avd.natural_sort` should be shorten with `ntp_server.nodes | arista.avd.natural_sort`

**Review by titom73 — CHANGES_REQUESTED** [review]
Thanks @colinmacgiolla for your PR.

Could you just fix opened comment and I will be happy to merge PR right away after you push the change.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ntp-servers.j2 [hunk 2]
```diff
@@ -9,15 +9,18 @@
...
+{%             set node_cli = node_cli ~ " " ~ node %}
+{%             if node == firstEntry  %}
+{%                set node_cli = node_cli ~ " prefer" %}
+{%             endif %}
 {{ node_cli }}
-{%     endfor %}
-{% endif %}
+{%         endfor %}
+{%     endif %}
+{% endif %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ntp-servers.j2]: missing newline at the end of the file

**Comment by colinmacgiolla** [comment]
Based on the [conversation here](https://github.com/aristanetworks/ansible-avd/issues/1091) I've started a new branch with an updated schema, but its not ready for pull yet

**Comment by ClausHolbechArista** [comment]
@colinmacgiolla should we move this PR to draft then until we know how the other one moves along? No need to clutter release notes etc with multiple overlapping changes.

**Comment by ClausHolbechArista** [comment]
Closed, replaced by #1137

**Pull request closed without merging**